### PR TITLE
chore: raise file not found for images

### PR DESCRIPTION
**Pull request by qued**
When trying to open an image in the `from_image_file` method, trying to open an empty filename would produce some non-helpful error, when ideally it should raise `FileNotFoundError`. This PR ensures that a `FileNotFoundError` is raised.

#### Testing:
With `main` branch active, run:
```python
from unstructured_inference.inference.layout import DocumentLayout

DocumentLayout.from_image_file("")

```
Observe that you get an `AttributeError`. Run the code again from this branch, observe that you get a `FileNotFoundError`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
-## 0.2.4-dev0
+## 0.2.4-dev1
 
+* Clarify error when trying to open file that doesn't exist as an image
```
Comment (cragwolfe) [inline CHANGELOG.md:3]: please delete the next line CHANGELOG line :)

File: unstructured_inference/inference/layout.py [hunk 2]
```diff
@@ -76,7 +77,13 @@ def from_file(cls, filename: str, model: Optional[Detectron2LayoutModel] = None)
     def from_image_file(cls, filename: str, model: Optional[Detectron2LayoutModel] = None):
         """Creates a DocumentLayout from an image file."""
         logger.info(f"Reading image file: {filename} ...")
-        image = Image.open(filename)
+        try:
+            image = Image.open(filename)
+        except Exception as e:
+            if os.path.isdir(filename) or os.path.isfile(filename):
```
Comment (MthwRobinson) [inline unstructured_inference/inference/layout.py:83]: If it's a directory, should we tell them it's a directory?

File: test_unstructured_inference/inference/test_layout.py [hunk 3]
```diff
@@ -267,3 +267,8 @@ def mock_get_elements(self, *args, **kwargs):
         .elements
     )
     assert elements[0] == mock_page_layout
+
+
+def test_from_image_file_raises_with_empty_fn():
+    with pytest.raises(FileNotFoundError):
```
Comment (MthwRobinson) [inline test_unstructured_inference/inference/test_layout.py:273]: Should we also have a test for the case where the user passes in a directory?

**Review by MthwRobinson — APPROVED** [review]
Two minor/optional suggestions. LGTM once you take a look at those!

File: unstructured_inference/inference/layout.py (hunk 2 above)
Comment (qued) [reply unstructured_inference/inference/layout.py:83]: From my testing passing a directory in this context gets handled properly by `Image.open`, raising an `IsADirectoryError`. I'll add a test case for this though.

File: test_unstructured_inference/inference/test_layout.py (hunk 3 above)
Comment (qued) [reply test_unstructured_inference/inference/test_layout.py:274]: Case added in [ff0f5db](https://github.com/Unstructured-IO/unstructured-inference/pull/19/commits/ff0f5db55d0482be82f3ae71422bffde5bad163e).

**Pull request merged**